libs/platform/src/lib/components/form/form-generator/form-generator/form-generator.component.ts

```diff
@@ -0,0 +1,217 @@
+
+    /**
+     * @description Event which notifies parent component that the form has been successfuly validated
+     * and submitted.
+     */
+    @Output()
+    formSubmitted = new EventEmitter<DynamicFormValue>();
+
+    /**
+     * @description Represents the form instance. @see NgForm
+     */
+    @ViewChild(NgForm) ngForm: NgForm;
+
+    /**
+     * @description Dynamically generated form. @see FormGeneratorService
+     */
+    form: FormGroup;
+    /**
+     * @description List of the form controls.
+     */
+    formControlItems: DynamicFormControl[];
+
+    /**
+     * Set of flags representing if particular form item should be visible to the user.
+     */
+    shouldShowFields: {[key: string]: boolean} = {};
+
+    /**
+     * Flag representing that form is loading
+     */
+    formLoading = true;
+
+    /**
+     * @hidden
+     */
+    private _allowSubscribe = true;
+
+    /**
+     * @hidden
+     */
+    private _formItems: DynamicFormItem[];
+
+    /**
+     * @hidden
+     */
+    private _formValueSubscription: Subscription;
+
+    constructor(
+        private _fgService: FormGeneratorService,
+        private _cd: ChangeDetectorRef
+    ) {
+    }
+
+    /**
+     * @hidden
+     */
+    ngOnDestroy(): void {
+        this._allowSubscribe = false;
+    }
+
+    /**
+     * @hidden
+     */
+    async _onSubmit(): Promise<void> {
+
+        this.form.markAllAsTouched();
+        this._cd.markForCheck();
+
+        // stop here if form is invalid
+        if (this.form.invalid || this.form.pending) {
+            return;
+        }
+
+        const formValue = await this._fgService.getFormValue(this.form);
+
+        this.formSubmitted.emit(formValue);
+
+        this.form.markAsPristine();
+    }
+
+    /**
+     * @hidden
+     */
+    _getErrors(errors: {[key: string]: any}): {type: string; value: any}[] {
+        return Object.entries(errors).map(e => {
+            const errorType = e[0];
+            let errorValue = e[1];
+
+            const defaultErrorValue = this._fgService.getValidationErrorHints(errorType);
+
+            if (defaultErrorValue) {
+                errorValue = defaultErrorValue;
+            }
+            return {type: errorType, value: errorValue};
+        });
+    }
+
+    /**
+     * @hidden
+     */
+    private async _generateForm(): Promise<void> {
+
+        this.formLoading = true;
+
+        const form = await this._fgService.generateForm(this.formItems);
+
+        this._formValueSubscription?.unsubscribe();
+
+        this.form = form;
+
+        this.formControlItems = Object.values(this.form.controls) as DynamicFormControl[];
+
+        this.shouldShowFields = await this._fgService.checkVisibleFormItems(this.form);
+
+        this._formValueSubscription = this.form.valueChanges
+        .pipe(takeWhile(() => this._allowSubscribe))
+        .subscribe(async () => {
+            this.shouldShowFields = await this._fgService.checkVisibleFormItems(this.form);
```

Comment:
this might be a problem cause valueChanges event is going to happen frequently, doing `detectChanges` everytime might cause a performance issues

Reply:
Added debounceTime and changed detectChanges to markForCheck